app/models/stream_item.rb

```diff
@@ -26,7 +26,11 @@ def self.shared_with(person)
       .includes(:person, :group)
       .where(streamable_type: shared_streamable_types)
       .where(shared: true)
-      .where("(group_id is not null or streamable_type != 'Message')")
+      .where(
+        "streamable_type != 'Message' or " \
+        "streamable_id in (:message_ids)",
+        message_ids: stream_item_message_ids(person)
+      )
```

Comment:
This is kind of a big change that I need to consider -- I think most people would be surprised to see private messages in their stream/timeline. I know they would only show for the sender or the recipient, but I don't think users will understand that intuitively (they will think others can see the same timeline).

Can you pull this change into a separate PR so we can discuss separately from the really cool feature you're building here?